src/esw/cameras.py

```diff
@@ -0,0 +1,351 @@
+#!/usr/bin/env python3
+
+from mrover.msg import CameraCmd
+from typing import List, Dict, Tuple
+
+import threading
+
+import rospy
+from mrover.srv import (
+    ChangeCameras,
+    ChangeCamerasRequest,
+    ChangeCamerasResponse,
+)
+
+import jetson.utils
+
+
+class VideoDevices:
+
+    device: int
+    resolution: List[str]
+    video_source: jetson.utils.videoSource
+    # Screens video source is displayed on
+    output_by_endpoint: Dict[str, jetson.utils.videoOutput]
+
+    def __init__(self, device: int):
+        self.resolution = []
+        self.device = device
+        self.video_source = None
+        self.output_by_endpoint = {}
+
+    def remove_endpoint(self, endpoint: str) -> None:
+        """
+        Removes screen endpoint from video source list.
+        If endpoint dictionary is empty, it also deletes video source.
+        :param endpoint: the endpoint (e.g. 10.0.0.7:5000)
+        """
+        assert endpoint in self.output_by_endpoint.keys()
+        del self.output_by_endpoint[endpoint]
+        if len(self.output_by_endpoint) == 0:
+            self.video_source = None
+
+    def create_stream(self, endpoint: str, args: List[str]) -> None:
+        """
+        Adds endpoint to video source list and creates video_source if not existing.
+        :param endpoint: the endpoint (e.g. 10.0.0.7:5000)
+        :param args: a list of strings that are the arguments
+        """
+
+        assert not (endpoint in self.output_by_endpoint.keys())
+
+        if self.is_streaming():
+            # It exists and another stream is using it
+            # TODO - make sure that this check does what we want to do (self.resolution != args)
+            if self.resolution != args:
```

Comment:
I think we need to reconcile the naming convention. Either use args for everything or resolution for everything if they are interchangeable

Reply:
done